noxfile.py

```diff
@@ -83,20 +86,71 @@ def fmt_py(session: nox.Session):
     _run(session, "black", ".", "--check")
 
 
-@nox.session(venv_backend="none")
-def clippy(session: nox.Session) -> None:
-    for feature_set in ["full", "abi3 full"]:
-        _run(
-            session,
-            "cargo",
-            "clippy",
-            f"--features={feature_set}",
-            "--all-targets",
-            "--workspace",
-            "--",
-            "--deny=warnings",
-            external=True,
-        )
+@nox.session(name="clippy", venv_backend="none")
+def clippy(session: nox.Session) -> bool:
+    if not _clippy(session):
+        session.error("one or more jobs failed")
+
+
+def _clippy(session: nox.Session, *, env: Dict[str, str] = None) -> bool:
+    success = True
+    env = env or os.environ
+    for feature_set in _get_feature_sets():
+        command = "clippy"
+        extra = ("--", "--deny=warnings")
+        if _get_rust_version()[:2] == (1, 48):
+            # 1.48 crashes during clippy because of lints requested
+            # in .cargo/config
+            command = "check"
```

Comment:
I did wonder about this; the speed difference wasn't much from what I saw, and it seems better to lint everything just in case. If this turns out to be not the case, we can revisit later.

Reply:
Agreed, if there is no measurable difference, then running Clippy wherever possible seems preferable.

Reply:
It was about 1s per command on my laptop, so measurable, but not large enough for me to really mind (although it does add up slightly over the N commands we run here).